src/mmw/apps/bigcz/clients/usgswqp/search.py

```diff
@@ -0,0 +1,129 @@
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql, [geojson])
+
+        return ';'.join([row[0] for row in cursor.fetchall()])
+
+
+def parse_geom(record):
+    lat = float(record['LatitudeMeasure'])
+    lng = float(record['LongitudeMeasure'])
+    return Point(lng, lat)
+
+
+def parse_record(record):
+    geom = parse_geom(record)
+
+    links = []
+    monitoring_description = record['MonitoringLocationDescriptionText']
+
+    return USGSResource(
+        id=record['MonitoringLocationIdentifier'],
+        title=record['MonitoringLocationName'],
+        description=monitoring_description,
+        author=None,
+        links=links,
+        created_at=None,
+        updated_at=None,
+        geom=geom,
+        details_url='https://www.waterqualitydata.us/provider/{prov}/{org}/{id}/'.format(prov=record['ProviderName'],  # NOQA
+                                                                                         org=record['OrganizationIdentifier'],  # NOQA
+                                                                                         id=record['MonitoringLocationIdentifier']),  # NOQA
+        sample_mediums=None,
+        variables=None,
+        service_org=record['OrganizationIdentifier'],
+        service_orgname=record['OrganizationFormalName'],
+        service_code=record['MonitoringLocationIdentifier'],
+        service_url='https://www.waterqualitydata.us/data/Result/search?siteid={}&mimeType=csv&sorted=no&zip=yes'.format(record['MonitoringLocationIdentifier']),  # NOQA
+        service_title=None,
+        service_citation='We thank the water quality portal.',
+        begin_date=None,
+        end_date=None,
+        monitoring_type=record['MonitoringLocationTypeName'],
+        provider_name=record['ProviderName']
+    )
+
+
+def search(**kwargs):
+    bbox = kwargs.get('bbox')
+    # Currently not being used
+    # to_date = kwargs.get('to_date')
+    # from_date = kwargs.get('from_date')
+
+    if not bbox:
+        raise ValidationError({
+            'error': 'Required argument: bbox'})
+
+    bbox_area = bbox.area() * SQKM_PER_SQM
+
+    if bbox_area > USGS_MAX_SIZE_SQKM:
+        raise ValidationError({
+            'error': 'The selected area of interest with a bounding box of {} '
+                     'km² is larger than the currently supported maximum size '
+                     'of {} km².'.format(round(bbox_area, 2), USGS_MAX_SIZE_SQKM)})  # NOQA
+    params = {
+        # bBox might be used in the future
+        # 'bBox': '{0:.3f},{1:.3f},{2:.3f},{3:.3f}'.format(bbox.xmin, bbox.ymin, bbox.xmax, bbox.ymax),  # NOQA
+        'huc': unique_huc12s_in(kwargs.get('geojson')),
+        'mimeType': 'csv',
+        'sorted': 'no',
+        'minresults': '1',
+        'zip': 'yes'
+    }
+
+    try:
+        response = requests.get(CATALOG_URL, params=params)
+        with ZipFile(BytesIO(response.content)) as z:
+            data = csv.DictReader(z.open(z.filelist[0].filename))
```

Comment:
I did some tests, and while some AoIs worked, others did not. Upon investigation I discovered that the `csv.DictReader` does not natively support Unicode data (see here: https://stackoverflow.com/questions/5004687/python-csv-dictreader-with-utf-8-data)

So I replaced it with a custom `read_unicode_csv` utility function:

```diff
diff --git a/src/mmw/apps/bigcz/clients/usgswqp/search.py b/src/mmw/apps/bigcz/clients/usgswqp/search.py
index b9de48d1..ea7c03ed 100644
--- a/src/mmw/apps/bigcz/clients/usgswqp/search.py
+++ b/src/mmw/apps/bigcz/clients/usgswqp/search.py
@@ -3,7 +3,6 @@ from __future__ import (absolute_import,
                         division,
                         print_function,
                         unicode_literals)
-import csv
 import requests
 
 from datetime import date
@@ -16,7 +15,7 @@ from django.contrib.gis.geos import Point
 from django.db import connection
 
 from apps.bigcz.models import ResourceList
-from apps.bigcz.utils import RequestTimedOutError
+from apps.bigcz.utils import read_unicode_csv, RequestTimedOutError
 
 from apps.bigcz.clients.usgswqp.models import USGSResource
 
@@ -113,7 +112,7 @@ def search(**kwargs):
     try:
         response = requests.get(CATALOG_URL, params=params)
         with ZipFile(BytesIO(response.content)) as z:
-            data = csv.DictReader(z.open(z.filelist[0].filename))
+            data = read_unicode_csv(z.open(z.filelist[0].filename))
     except requests.Timeout:
         raise RequestTimedOutError()
 
diff --git a/src/mmw/apps/bigcz/utils.py b/src/mmw/apps/bigcz/utils.py
index 878df2d0..555386ba 100644
--- a/src/mmw/apps/bigcz/utils.py
+++ b/src/mmw/apps/bigcz/utils.py
@@ -3,6 +3,8 @@ from __future__ import print_function
 from __future__ import unicode_literals
 from __future__ import division
 
+import csv
+
 from apps.bigcz.models import BBox
 
 from rest_framework.exceptions import APIException
@@ -51,3 +53,12 @@ class RequestTimedOutError(APIException):
     status_code = 408
     default_detail = 'Requested resource timed out.'
     default_code = 'request_timeout'
+
+
+def read_unicode_csv(utf8_data, **kwargs):
+    csv_reader = csv.DictReader(utf8_data, **kwargs)
+    for row in csv_reader:
+        yield {
+            unicode(key, 'utf-8'): unicode(value, 'utf-8')
+            for key, value in row.iteritems()
+        }
```